packages/flutter/lib/src/widgets/router.dart

```diff
@@ -286,11 +315,38 @@ class Router<T> extends StatefulWidget {
     required this.routerDelegate,
     this.backButtonDispatcher,
     this.restorationScopeId,
-  })  : assert(
-          (routeInformationProvider == null && restorationScopeId == null) || routeInformationParser != null,
```

Comment:
A router can be restored with just routerDelegate

Reply:
In that cases, there is really nothing need to be restored because all the state will be in the routerDelegate